accounts/abi/bind/backends/simulated_test.go

```diff
@@ -1114,3 +1114,41 @@ func TestSimulatedBackend_CallContractRevert(t *testing.T) {
 		sim.Commit()
 	}
 }
+
+func TestSimulatedBackend_Reorg(t *testing.T) {
+	testAddr := crypto.PubkeyToAddress(testKey.PublicKey)
+	sim := simTestBackend(testAddr)
+	defer sim.Close()
+	bgCtx := context.Background()
+
+	// Mine 5 blocks.
+	for i := uint64(0); i < 5; i++ {
+		sim.Commit()
+	}
+	// Assert chain length.
+	if sim.blockchain.CurrentBlock().NumberU64() != 5 {
+		t.Error("wrong length of chain")
+	}
+	// Parent of the reorg is at number 0.
+	parent, err := sim.BlockByNumber(bgCtx, big.NewInt(0))
+	if err != nil {
+		t.Errorf("could not retrieve block: %v", err)
+	}
+	// Add two blocks via a reorg onto parent.
+	if err := sim.Reorg(bgCtx, 4, 6, func(txs []types.Transactions) []types.Transactions { return txs }); err != nil {
+		t.Errorf("could not reorg: %v", err)
+	}
+	// Get the block directly ontop of the parent.
+	newChain, err := sim.BlockByNumber(bgCtx, big.NewInt(1))
+	if err != nil {
+		t.Errorf("could not retrieve block: %v", err)
+	}
+	// Check that the new chain is based on parent.
+	if newChain.ParentHash() != parent.Hash() {
+		t.Error("wrong parent hash in newChain")
+	}
+	// Assert chain length.
+	if sim.blockchain.CurrentBlock().NumberU64() != 7 {
+		t.Error("wrong length of new chain")
+	}
```

Comment:
Could you either add a second test, or extend this one to actually shuffle some transactions? 
Would be nice to see that this functionality works not just with empty blocks

Reply:
Yes currently it is a bit boring :laughing:. I will try to add a contract test.